### ISSUE TITLE
UI - Tab View: Oddities Choosing Projects in Alternate Tab

### ISSUE THREAD

**Issue by 3DEsprit** [Microsoft/vscode#47926]
Issue Type: <b>Bug</b>

Working: 1.22.1
Broken: 1.22.2

OS: OSX 10.13.4

**Issue**: When choosing a project for a new tab or tabs, instead of setting the folder to the tab it was opened from, the project chosen may replace another tab's project, clear the project from another tab, open a whole new tab, or close multiple tabs. For some reason this seems to happen with recent projects, but after much testing seems to happen with any folder. Just to clarify, these are NOT saved workspaces, simply project folders. 

1. Switch to Tab Overview, create a new Tab, and select that tab editor.
2. Open a recent project folder from the _File_ menu, or _Open Folder_ button in the navigation sidebar
3. Depending on what folder you chose results may vary, but the majority of the time a folder is not opened as expected. 

To better show this issue, here are some screencasts showing this behavior:

https://d.pr/i/zJEL1y

https://d.pr/i/4soWRV

https://d.pr/i/uuHkg2

**Additional Details**

In these screencasts I am choosing a few recent projects that showed this behavior, but each time they did something different. You can see where switching a project will actually remove a project from one of the tabs returning me to the welcome tab, closing several others at a time, almost never showing in the tab I opened from. 

**Summary**

This was working earlier today with 1.22.1, and after I upgraded to 1.22.2 I found myself unable to set my projects in these multiple tabs. Just to make sure I disabled all installed extensions, but the behavior persisted. 

Prior to this upgrade I was able to select whatever project I wanted in a tab, and it would become the dedicated project for that editor without anything changing it when setting other tabs.  I checked through the release notes to see if there was anything that may have changed, but did not find anything relevant to tabs. 

VS Code version: Code 1.22.2 (3aeede733d9a3098f7b4bdc1f66b63b0f48c1ef9, 2018-04-12T16:32:53.389Z)
OS version: Darwin x64 17.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz (8 x 2600)|
|Load (avg)|3, 5, 4|
|Memory (System)|16.00GB (5.61GB free)|
|Process Argv|/Applications/Visual Studio Code.app/Contents/MacOS/Electron|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (26)</summary>

Extension|Author (truncated)|Version
---|---|---
react-native-react-redux|EQu|1.8.0
vscode-docker|Pet|0.0.26
unity-tools|Tob|0.1.9
unity-debug|Uni|1.3.0
shaderlabvscodefree|aml|1.1.0
vscode-processes|and|0.0.4
vscode-custom-theme|ben|1.1.3
solargraph|cas|0.15.0
vscode-eslint|dba|1.4.8
githistory|don|0.4.0
vscode-babel-coloring|dza|0.0.4
prettier-vscode|esb|1.3.0
vscode-install-vsix|fab|1.1.1
flow-for-vscode|flo|0.8.0
sass-lint|gle|1.0.1
rainbow-csv|mec|0.2.1
vscode-language-babel|mgm|0.0.14
azure-account|ms-|0.4.0
csharp|ms-|1.14.0
node-debug2|ms-|1.19.0
debugger-for-chrome|msj|4.3.0
Ruby|reb|0.18.0
vscode-icons|rob|7.22.0
sass-indented|rob|1.4.9
react-beautify|tai|0.3.0
vscode-react-native|vsm|0.6.6

(2 theme extensions excluded)

</details>

**Comment by 3DEsprit** [comment]
Still a different issue than what the bot thought was duplicate. 

I was able to download a version of 1.21.1 to regain this feature, but keeping the 1.22.2 for any future tests.

**Comment by bpasero** [comment]
@3DEsprit we currently do not support to open the same folder in multiple windows, it seems to me the folder is already opened in another window and we just switch focus to it, can you confirm?

**Comment by 3DEsprit** [comment]
I could see where that would seem to be the issue with choosing between a small set of projects in the screencasts, but the issues I'm trying to portray is what happens when trying to assign a single unopened project to a tab even when not already loaded. It was nearly impossible with 1.22.2 to choose a tab, and set a project to that tab, regardless of it being open. 

To give a bit more context, here is what the workflow was like, and what it is like in 1.22.2:

_with freshly opened editor (single tab)_

1. Load Project 1 into first tab
2. Open tab overview, and create a 2nd tab
3. Load Project 2 into new tab
4. Open tab overview, and create a 3rd tab
5. Load Project 3 into new tab

What is happening with 1.22.2

1. Load Project 1 into first tab
2. Open tab overview, and create 2nd tab
3. Load Project 2 from 2nd tab, but opens in 1st tab
4. Load Project 1 back into 1st tab
5. Load Project 3 into 2nd tab, but loads in 1st tab, and closes 2nd tab

Coming back to this today, I'm starting to wonder if this was some sort of artifact from the state prior to the upgrade, as this is stopped happening after starting up and using 1.21.1 (as if it updated session settings). 

To give more context of what happened during the upgrade from 1.22.1 to 1.22.2, I had 4 tabs open before it closed itself, and reopened up with a single tab (expected), but still showed the need for an upgrade. It wasn't until upgrading the 2nd time that it did upgrade to 1.22.2, and reopened with the welcome. The projects that were showing the most trouble were those open before the upgrade.

My end goal is to always open 3-4 different projects in tabs of their own (which is easily accomplished with 1.21.1 and 1.22.1), as some of my plugins will not work when there are multiple projects in a workspace.

**Comment by bpasero** [comment]
@3DEsprit when you talk about "tabs" you actually mean "windows" right? You want to open 4 different folders in 4 different windows?

Does it reproduce if you shutdown all instances and start fresh? And if you start fresh, this should work:
* open the first instance
* select File > New Window
* select File > Open Folder
* pick the folder (a folder that is NOT already opened)

=> does this folder open in the second window that you opened as part of clicking "File > New Window"?

**Comment by 3DEsprit** [comment]
Correct, tabbed editors, windows, whatever you want to call them. I'm just trying to stay as close to the terminology used by the editor, and what I recollect when the feature was added.

While I did not try the above process exactly, I did noticed that opening projects I hadn't opened recently worked as intended without issue. 

Looking over the screencasts I do see I wasn't overlapping projects that were already open in a tabbed editor/window (watch the project being opened and what shows as open). I can see where it would look like the same project being opened, especially after opening one project, and then opening again because it didn't show loaded in any of the tabbed editors. It may also be worth pointing out there are 3 projects I'm opening that are similarly named (barista-core, barista-agent, barista-runner), which may have been perceived as a single project at a glimpse; 3 projects that live in my tab view each in their own editor.

Unfortunately I am unable to reproduce this any longer as after using 1.21.1 this stopped in the 1.22.2 (I should have saved the preferences, that's my bad), and now works as intended, of course I can always try to reproduce the entire process again if I could find a download of 1.22.1. 

With the results and resolve, could this simply be chalked up to a corrupted restore state?
